0018-connector-risk-mitigations/0018-connector-risk-mitigations.md

```diff
@@ -17,25 +17,28 @@ The main risk connectors face in Interledger is being unable to fulfill the inco
 
 * **Managing "Payment Bandwidth"** - Connectors should monitor the percentage of their total liquidity that is on hold at any given time for a given peer, customer, or destination prefix and may reject incoming payment requests if that party exceeds their allocated "payment bandwidth". Connectors may allocate less bandwidth to unknown or untrustworthy senders or receivers. Note that ILP Packets contain the destination account but not the source account, so connectors should apply this logic to the immediate peer or customer that they receive the payments from and possibly the destination account or prefix.
-* **Preferring Smaller Payments** - Smaller payments place a smaller percentage of the connector's liquidity on hold and thus each one carries less risk that the payment being prepared and then failing would create significant opportunity cost
+* **Preferring Smaller Payments** - Smaller payments place a smaller percentage of the connector's liquidity on hold and thus each one carries less risk that the payment being prepared and then failing would create significant opportunity cost - in ILPv4, this is the default.
 * **Blacklisting Senders and Receivers** - Connectors may refuse to facilitate payments from certain sources or to certain destinations if their rate of failed payments is unusually high
 
+In the worst case, during the fulfill phase a connector along the path fulfilling the payment could withhold the fulfill packet such that their corresponding connector would not receive a payment. The corresponding connector would at worst lose the unit amount being streamed in a single packet (i.e. 1 satoshi). The fulfilling connector would gain the unit amount and would put their reputation as a connector on the line and would miss out on future fee gains. Additionally, the fulfilling connector would have to close the payment channel between them and the corresponding connector, as the corresponding connector would choose not to facilitate payments through the channel after being cheated. This would result in an on-chain transaction which would make the economics for the attacker not worth it, as they could be making more by continuing to facilitate fees.
+
 ## Exchange Rate Volatility
 
-Once connectors prepare their outgoing transfers, they are committed to the payment even if the exchange rate between the assets fluctuates.
+Once connectors prepare their outgoing transfers, they are committed to the payment even if the exchange rate between the assets fluctuates. In ILPv4, the support of low-value packets allows for connectors to do exchange rates on a per-packet basis. By having shorter timeouts for the prepare packets, senders have stronger guarantees for quick resolution of their packets and connectors are limiting their risk of being taken advantage from by the "free option problem" standpoint (ensure that packets are held until just before their expiry and then fulfilled only if the exchange rate moved in the sender's favor).
```

Comment:
I'm not sure this section is super relevant anymore. If you have packets with very short timeouts (and relatively low amounts of money), this kind of becomes a non-issue for the connector

Reply:
@emschwartz I wonder if it's better to take things out or leave them as non-issues somewhere in the RFCs. Some people probably have questions as to whether ILP is susceptible to such things based on looking through other protocols and I think having answers to those type of questions is good for legitimacy

Reply:
Sure. This description sounds off to me.

> In ILPv4, the support of low-value packets allows for connectors to do exchange rates on a per-packet basis.

You can always set rates on a per-packet basis, it's just that the packet amounts are lower now and more importantly, the time you're holding it for is shorter.

> senders have stronger guarantees for quick resolution of their packets

This feels out of place in a doc on connector risk mitigation

> "free option problem" standpoint (ensure that packets are held until just before their expiry and then fulfilled only if the exchange rate moved in the sender's favor).

Trying to introduce this problem and say that it's solved in one sentence is a bit rushed. If you want to include it, it would be better to have an explanation that's a little longer and clearer (or link to another description of that issue).

Reply:
> senders have stronger guarantees for quick resolution of their packets
>> This feels out of place in a doc on connector risk mitigation

Agreed, I think this was because of my misunderstanding about payments above.